guides/source/active_record_migrations.md

```diff
@@ -967,36 +959,34 @@ end
-
-NOTE: `db/schema.rb` cannot express database specific items such as triggers,
-sequences, stored procedures or check constraints, etc. Please note that while
-custom SQL statements can be run in migrations, these statements cannot be reconstituted
-by the schema dumper. If you are using features like this, then you
-should set the schema format to `:sql`.
-
-Instead of using Active Record's schema dumper, the database's structure will
-be dumped using a tool specific to the database (via the `db:structure:dump`
-rails task) into `db/structure.sql`. For example, for PostgreSQL, the `pg_dump`
-utility is used. For MySQL and MariaDB, this file will contain the output of
-`SHOW CREATE TABLE` for the various tables.
-
-Loading these schemas is simply a question of executing the SQL statements they
-contain. By definition, this will create a perfect copy of the database's
-structure. Using the `:sql` schema format will, however, prevent loading the
-schema into a RDBMS other than the one used to create it.
+on.
+
+`db/schema.rb` cannot express everything your database may support such as
+triggers, sequences, stored procedures or check constraints, etc. Please note
+that while custom SQL statements can be run in migrations, these statements
+cannot be reconstituted by the schema dumper. If you are using features like
+this, you can set the schema format to `:sql` or you can use `db/schema.rb`
+solely as a reference. In this case you would create a new instance of your
+database by running your migrations from scratch instead of running `rails
+db:schema:load`
+
+When the schema format is set to `:sql`, the database structure will be dumped
+using a tool specific to the database into `db/structure.sql`. For example, for
+PostgreSQL, the `pg_dump` utility is used. For MySQL and MariaDB, this file will
+contain the output of `SHOW CREATE TABLE` for the various tables.
+
+Loading these `db/structure.sql` is done by executing the SQL statements it
+contains. By definition, this will create a perfect copy of the database's
+structure. However, `db/structure.sql` tends to be harder to use as a reference
+during development because it is not as consice.
 
 ### Schema Dumps and Source Control
 
-Because schema dumps are the authoritative source for your database schema, it
-is strongly recommended that you check them into source control.
+Because schema files are commonly used to create new databases, It is strongly
+recommended that you check your schema file into source control.
 
-`db/schema.rb` contains the current version number of the database. This
-ensures conflicts are going to happen in the case of a merge where both
-branches touched the schema. When that happens, solve conflicts manually,
-keeping the highest version number of the two.
+Merge conflicts can occur in your schema file when two branches modify schema.
+To resolve these conflicts run `rails db:migrate` to regenerate the schema file.
```

Comment:
Why would we recommend that to run migrations again when this conflict can be easily resolved manually?

Reply:
I've always found it easier to re-run the migrations than edit the file by hand, which is probably just personal preference. However, it can also avoid mistaken conflict resolution in the case where the migrations effect the same column in a table (changing default, changing null constraint, for example) and also abides by the request at the top of the generated file to not edit by hand.

Reply:
You're also going to need to run the migrations you're pulling in from the other branch at some point, so why not now?